scalameta/src/main/scala/scala/meta/internal/parsers/Parsers.scala

```diff
@@ -562,6 +562,18 @@ private[meta] class Parser(val input: Input)(implicit val dialect: Dialect) { pa
     }
   }
 
+  // corresponding quasiquote (rank 1) for expressions like `{..$smth}`
+  def constructR1Quasi[T <: Tree : AstMetadata](ctor: => Tree): List[T] = {
+    val tree = {  
+      val q = ctor match {
+        case q: Quasi if q.rank == 0 => q
+        case _ => syntaxError("impossible to construct rank-1 quasiquote", at = token)
+      }
+      atPos(q, q)(implicitly[AstMetadata[T]].quasi(q.rank, q.tree))
+    }
+    implicitly[AstMetadata[T]].quasi(1, tree) :: Nil
+  }
+
```

Comment:
Consider `cases ++= constructR1Quasi[Case](expr())`. Here `expr` is `Term`, so is `T`. But `Case` is not `Term`, and that need to be explicitly stated.

Here `expr()` is a way to get ellipsis' variable name (e.g. `..$smth` => `smth`), and `Case` is a type for final Quasi.